packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -36,21 +37,22 @@ import { createStateSymbol, reportDiagnostic } from "./lib.js";
 import { getAllModels, getSdkEnum, getSdkModel } from "./types.js";
 
 export const namespace = "Azure.ClientGenerator.Core";
+const AllScopes = Symbol.for("@azure-core/typespec-client-generator-core/all-scopes");
 
 function getScopedDecoratorData(context: SdkContext, key: symbol, target: Type): any {
-  const retval: Record<string, any> = context.program.stateMap(key).get(target);
+  const retval: Record<string | symbol, any> = context.program.stateMap(key).get(target);
```

Comment:
```suggestion
  const retval: Record<string | typeof AllScopes, any> = context.program.stateMap(key).get(target);
```

that way only that symbol is allowed